core-bundle/assets/controllers/check-all-controller.js

```diff
@@ -3,7 +3,61 @@ import { Controller } from '@hotwired/stimulus';
 export default class extends Controller {
     static targets = ['source', 'input'];
 
-    update() {
+    static values = {
+        rowGuards: {
+            type: Array,
+            default: ['input', 'label', 'button', 'a', '.operations'],
+        },
+    };
+
+    initialize() {
+        this.start = null;
+    }
+
+    #shiftToggle(el) {
+        const thisIndex = this.inputTargets.indexOf(el);
+        const startIndex = this.inputTargets.indexOf(this.start);
+
+        const from = Math.min(thisIndex, startIndex);
+        const to = Math.max(thisIndex, startIndex);
+        const status = this.start.checked;
+
+        for (let i = from; i <= to; i++) {
+            this.inputTargets[i].checked = status;
+        }
+    }
+
+    toggleRow(event) {
+        if (event.target.closest(this.rowGuardsValue.join(','))) {
+            return;
+        }
+
+        const target = event.currentTarget.querySelector('[data-contao--check-all-target="input"]');
```

Comment:
Just out of my head, but this should not hardcode the controller name.

```suggestion
        const target = event.currentTarget.querySelector(`[data-${this.identifier}-target="input"]`);
```

Reply:
See d76f04f45951b094bbd6e5912c7466971f958812